app/controllers/volunteer_ops_controller.rb

```diff
@@ -7,9 +7,10 @@ class VolunteerOpsController < ApplicationController
 
   def search
     @query = params[:q]
-    @volunteer_ops = VolunteerOp.order_by_most_recent.search_for_text(@query)
+    @volunteer_ops = VolunteerOp.order_by_most_recent.search_for_text(@query).to_a
     flash.now[:alert] = SEARCH_NOT_FOUND if @volunteer_ops.empty?
-    @markers = BuildMarkersWithInfoWindow.with(VolunteerOp.build_by_coordinates, self)
+    @markers = BuildMarkersWithInfoWindow
+      .with(VolunteerOp.build_by_coordinates(@volunteer_ops), self)
```

Comment:
Hey @tansaku I think that you idea about passing the results of the query is great but I would like to analyze this a little further because we are only interested on building coordinates on those `@volunteer_ops` that have coordinates, and the way that we are working now is to trigger all the logic we have for any result of that query, when we could be working only with the ones that we now are gonna be grouped by coordinates, and those will conform the `@markers`. Right now if any result has coordinates we build {[nil, nil]:=> [vol_ops]}.